### PR TITLE
Handle inflect names

### PR DESCRIPTION
**Pull request by andrewcarretta**
## Description
The value returned from inflect can be a Boolean. This needs to be handled not returned as False
Credit @gthank and @stavvy-rotte for their other PR's

## Changes
- Reimplemented code from https://github.com/agronholm/sqlacodegen/pull/228
- Created tests to cover these scenarios

## Related
- https://github.com/agronholm/sqlacodegen/pull/228
- https://github.com/agronholm/sqlacodegen/pull/207

### CODE REVIEW

**Review by agronholm — CHANGES_REQUESTED** [review]
The new tests are almost identical. They could be parametrized (table name, class name, relationship name).

**Comment by andrewcarretta** [comment]
> hey could be parametrized (table name, class name, relationship name).

I made them separate because all your other tests are atomic / self-contained.

**Comment by agronholm** [comment]
If I'm correct about all of the tests being useful, then parametrizing is the way to go.

File: tests/test_generators.py [hunk 1]
```diff
@@ -1822,6 +1822,63 @@ class Singular(Base):
...
+    def test_use_inflect_plural_es(self, generator: CodeGenerator) -> None:
+        self.use_inflect_pluralization(generator, "status", "statuses")
+
+    @pytest.mark.parametrize("generator", [["use_inflect"]], indirect=True)
+    def test_use_inflect_plural_ies(self, generator: CodeGenerator) -> None:
+        self.use_inflect_pluralization(generator, "study", "studies")
+
+    @pytest.mark.parametrize("generator", [["use_inflect"]], indirect=True)
+    def test_use_inflect_singular(self, generator: CodeGenerator) -> None:
+        self.use_inflect_pluralization(generator, "moose", "moose")
```
Comment (agronholm) [inline tests/test_generators.py:1880]: This is not what I meant. I meant for you to use `@pytest.mark.parametrize` to supply those parameters.

Comment (andrewcarretta) [reply tests/test_generators.py]: Implemented parameterized tests

**Comment by andrewcarretta** [comment]
> The new tests are almost identical. They could be parametrized (table name, class name, relationship name).

Used your variable names for the parameterized tests

**Comment by agronholm** [comment]
Fix the pre-commit issues please. Then I can merge.

**Review by agronholm — APPROVED** [review]

**Comment by agronholm** [comment]
Thanks.

**Pull request merged**